cmake/modules/CheckCompiler.cmake

```diff
@@ -147,7 +147,7 @@ if(${CMAKE_CXX_COMPILER_ID} MATCHES Clang)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wc++11-narrowing -Wsign-compare -Wsometimes-uninitialized -Wconditional-uninitialized -Wheader-guard -Warray-bounds -Wcomment -Wtautological-compare -Wstrncat-size -Wloop-analysis -Wbool-conversion")
 elseif(CMAKE_COMPILER_IS_GNUCXX)
   if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 7)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-implicit-fallthrough -Wno-noexcept-type")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-implicit-fallthrough")
```

Comment:
Ah, I think I did overlook the fact that the warning about the unknown flag may be from dictionary generation by cling rather than by the compilation of ROOT with GCC7, is that correct? If yes, I can revert this commit. That said, there are no warnings due to noexcept types, so maybe disabling this warning isn't really necessary anyway.